server/src/main/java/io/crate/planner/optimizer/symbol/rule/SwapCastsInComparisonOperators.java

```diff
@@ -41,12 +46,53 @@
 import io.crate.types.DataType;
 import io.crate.types.DataTypes;
 
-
 public class SwapCastsInComparisonOperators implements Rule<Function> {
 
     private final Capture<Function> castCapture;
     private final Pattern<Function> pattern;
 
+    /**
+     * A cast is 'swappable' if the cast does not cause loss of precision. This method executes the casts in uncertain
+     * situations. i.e.:: casting 126 type of int to a byte, which is a narrowing conversion, is convertable since 126
+     * does fit in a byte.
+     */
+    private static boolean isSwappable(Function cmpOp) {
+        if (cmpOp.arguments().get(1) instanceof Literal<?> literal) {
+            Function cast = (Function) cmpOp.arguments().getFirst();
+            Reference ref = (Reference) cast.arguments().getFirst();
+            DataType<?> refInnerType = ArrayType.unnest(ref.valueType());
+            DataType<?> literalInnerType = ArrayType.unnest(literal.valueType());
+            if (!DataTypes.isNumeric(literalInnerType) || !DataTypes.isNumeric(refInnerType)) {
+                return true;
+            }
+            if (SAFE_CONVERSIONS.getOrDefault(literalInnerType.id(), Set.of()).contains(refInnerType)) {
```

Comment:
Maybe you can extract a method instead of directly using `SAFE_CONVERSIONS`.

Reply:
Thanks, updated.